### DISCUSSION TITLE
How do I use recaptcha v3?

### DISCUSSION THREAD

**Discussion by bluebamus** [django-recaptcha/django-recaptcha#314] [Q&A] [answered]
After setting foms.py like "captcha = ReCaptchaField(widget=ReCaptchaV3)"
I checked with the “Clean” function.
I showed the code like token
For example "...aWasdftFvasdfpCF0b2AHPqYocJBtCqgV4qgw_H8nasdfqzLGKdioRWTlAWlU5Lm5EZnmSGovvJI60ohbaU9N6asdfXe0z0AgGtc-dKNAHSu1bwgHWrXWEaGBRasdf8wMEI-QFjOMLxrcq3E- _s0 Dyasdaf4GrND0x7L6uHasdf7XJK7j9OCPAS7BkFLC4jI22JmmuPCsao1bcAHTuAmHQuxSZ8hs8tFdfYasfdN4jTLxIH-PpsKs_cmnyKWsdg4tQ69mZ-3MnSeWsd7IrGCw28Dophrnet0RdkeSm3HF v QKg4QbQbqTfxI0Q_91AbF47v2dLR..."
Is this the correct return value?
It is difficult because there are no examples of how to evaluate it.

**Comment by Stormheg** [comment, marked as answer]
Hi @bluebamus, you don't have to write any validation logic for the token. You only have to call `is_valid()` or `full_clean()` on your form, Django Recaptcha will do the heavy lifting for you!

Upon form submission, Django Recaptcha will look at the the token/field value and use it to check if your form was submitted by a bot. If it is a bot, a validation error will be raised for your ReCaptchaField.

Hope this answers your question. Let me know!
